cpp/src/copying/contiguous_split.cu

```diff
@@ -263,47 +263,51 @@ size_type compute_offset_stack_size(InputIter begin, InputIter end, int offset_d
  * compute_offset_stack_size in the case of nested types.
  */
 struct compute_offset_stack_size_functor {
-  template <typename T, std::enable_if_t<std::is_same<T, cudf::string_view>::value>* = nullptr>
+  template <typename T>
   size_type operator()(column_view const& col, int offset_depth)
   {
-    //     # of list columns above us    # offsets + validity
-    return ((offset_depth) * (1 + (col.nullable() ? 1 : 0))) +
-           //     # + local offsets             # chars column
-           ((offset_depth + 1) * (1));
+    //     # of list columns above us   # of buffers coming from this column
+    return offset_depth * (1 + (col.nullable() ? 1 : 0));
   }
+};
 
-  template <typename T, std::enable_if_t<std::is_same<T, cudf::list_view>::value>* = nullptr>
-  size_type operator()(column_view const& col, int offset_depth)
-  {
-    //     # of list columns above us    # offsets + validity
-    //       and ourself
-    return ((offset_depth) * (1 + (col.nullable() ? 1 : 0))) +
-           //     add children
-           compute_offset_stack_size(col.child_begin() + 1, col.child_end(), offset_depth + 1);
-  }
+template <>
+size_type compute_offset_stack_size_functor::operator()<cudf::string_view>(column_view const& col,
+                                                                           int offset_depth)
+{
+  //     # of list columns above us    # offsets + validity
+  return ((offset_depth) * (1 + (col.nullable() ? 1 : 0))) +
+         //     # + local offsets             # chars column
```

Comment:
The spacing in the comments doesn't line up with anything. You would be better off storing the values in clearly named variables and then doing the math using those clear names.

Reply:
Sorry. Clang format messing things up again.

Reply:
Why use comments to say what you can say with code?

Reply:
Comments are in English, which is inherently more expressive. And in this case, before clang-format messed things up, provided additional, visual context by illustrating what each part of the expression means.  
```
         // # of list columns above us        # offsets + validity
  return ((offset_depth)                 *    (1 + (col.nullable() ? 1 : 0))) +
         // # + local offsets                 # chars column
         ((offset_depth + 1)             *    (1));
```